cpu: moved all vendor headers to CPUNAME/include/vendor/

haukepetersen opened the pull request:
The reason for this is to simplify handling of vendor headers in the future, e.g. by easier excluding them from doxygen builds etc...

kaspar030 commented on doc/doxygen/riot.doxyfile:
```diff
@@ -818,30 +818,11 @@ EXCLUDE_SYMLINKS       = NO
-                         */cpu/lpc11u34/include/LPC11Uxx.h \
-                         */cpu/lpc1768/include/LPC17xx.h \
-                         */cpu/lpc*/include/core_cm*.h \
+                         */cpu/*/include/vendor/*
```
Why not ```*/vendor/*```?

haukepetersen replied:
good point. will adapt

kaspar030 commented:
Could you also adapt the cppcheck filters in ```dist/tools/cppcheck/check.sh```?

haukepetersen commented:
done

haukepetersen commented:
fixed excludes for externc and some empty line at EOF issues.

aabadie commented:
I'm globally +1 with the goal of this PR but as you have noticed, I have some opened PRs that will be impacted : #6651, #6025, #6553, #6594, #6615, #6625.
Some of them are opened for a long time now (#6025) or closed to being merged (just missing a positive review)  (#6651) .
Would it be possible to merge them first ?

gebart commented:
Good idea, but what about having two separate trees, instead of polluting the include tree. 
Like this:
Cpu/stm32f4/vendor/include

Also, while I can see that having to prefix vendor/ on all vendor includes has its merits with regards to readability, it may also cause a hassle when vendor headers include other vendor headers (e.g. Atmel CPUs). 
I'm writing this from the phone so I haven't really managed to get a good overview of this PR, so please excuse me if I have missed something here.

haukepetersen commented:
@aabadie: I would prefer to merge this PR first (as there are probably another 20 PRs or so, that would need to be merged first...), as right now there is the effort for the 'code quality task force'.

@gebart I think this PR is not `polluting the include tree`, IMHO it's quite the contrary by cleanly separating the vendor headers. Regarding the vendor libraries (as e.g. the Atmel files), this PR has no negative impact, as they use relative include paths and find their 'peer' files locally without even looking into the include tree...

gebart commented on doc/doxygen/riot.doxyfile:
```diff
@@ -818,30 +818,11 @@ EXCLUDE_SYMLINKS       = NO
-                         */cpu/lpc11u34/include/LPC11Uxx.h \
-                         */cpu/lpc1768/include/LPC17xx.h \
-                         */cpu/lpc*/include/core_cm*.h \
+                         */vendor/* \
```
nice cleanup here! :+1: 
Could you move it to the correct place to keep the list sorted?

gebart commented:
@haukepetersen After looking at this properly on my PC I am starting to agree that this is a good solution.
Added the major tag because this touches a lot of files across the tree.

@aabadie the rebase of other PRs touching these vendor headers should be quite uncomplicated. I agree with @haukepetersen that merging this is preferable compared to merging everything else first. There are new PRs opened all the time so it would be an infinite catching up to try to get this merged.

aabadie commented:
@gebart, indeed, just tested the rebase, things works fine (even if I have to modify files afterwards but that was expected anyways).

haukepetersen commented:
Addressed comment by @gebart above and squashed. Now waiting for the CI once more to make sure I didn't break anything and then we should be ready to merge this.

kaspar030 commented:
Murdock2 currently chokes as cppcheck takes longer than 5 minutes, which is Murdock2's default job timeout. The other CI's agree, so I'd say merge!
